### ISSUE TITLE
`num_latent` is not getting logged in wandb

### ISSUE THREAD

**Issue by epignatelli** [Kokcr3ator/latent-action-molecule-generation#23]
And maybe also other metrics. 

This is what we are logging right now:

<summary>
<details>
<pre>
{
  "log": {
    "value": {
      "name": "ppo_qed_controllable_nlatents8_envs16_steps256_seed42",
      "group": "ppo_controllable",
      "project": "interdiff",
      "_target_": "interdiff.logging.wandb_logger.WandbLogger"
    }
  },
  "ppo": {
    "value": {
      "lr": 0.000001,
      "budget": 10000000,
      "vf_coef": 0.5,
      "_target_": "interdiff.ppo.HParams",
      "clip_eps": 0.2,
      "discount": 1,
      "ent_coef": 0.01,
      "num_envs": 16,
      "save_dir": "ckpts/ppo_qed_controllable_nlatents8_envs16_steps256_seed42",
      "anneal_lr": true,
      "num_steps": 256,
      "gae_lambda": 0.95,
      "lambda_kld": 0.05,
      "num_epochs": 1,
      "num_actions": 500,
      "log_to_wandb": true,
      "random_start": false,
      "weight_decay": 0.01,
      "log_frequency": 1,
      "max_grad_norm": 1,
      "eval_frequency": 1,
      "clip_value_loss": true,
      "num_minibatches": 8,
      "max_episode_length": 128,
      "wandb_project_name": "interdiff",
      "normalise_advantage": true
    }
  },
  "sys": {
    "value": {
      "dtype": "float16",
      "device": "cuda",
      "compile": true,
      "always_save_checkpoint": false
    }
  },
  "ckpt": {
    "value": {
      "path": "ckpts/policydistillation_nlatents8_vocab500_seed42",
      "ckpt_name": "best.pt",
      "init_from": "resume"
    }
  },
  "data": {
    "value": {
      "name": "zinc",
      "smiles": "interdiff/data/zinc/zinc.txt",
      "pad_token_id": 0
    }
  },
  "seed": {
    "value": 42
  },
  "optim": {
    "value": {
      "beta1": 0.9,
      "beta2": 0.99
    }
  },
  "_wandb": {
    "value": {
      "m": [],
      "t": {
        "1": [
          1
        ],
        "2": [
          1
        ],
        "3": [
          13,
          55
        ],
        "4": "3.10.18",
        "5": "0.20.1",
        "12": "0.20.1",
        "13": "linux-x86_64"
      },
      "cli_version": "0.20.1",
      "python_version": "3.10.18"
    }
  },
  "loader": {
    "value": {
      "_target_": "interdiff.data.RLLoader.FinetuneControlable",
      "ckpt_name": "best.pt",
      "ckpt_controllable_path": "ckpts/pretrain_controllable_vocab500_nlatent8_seed42"
    }
  },
  "reward": {
    "value": {
      "task": "qed"
    }
  },
  "context": {
    "value": {
      "seq_len": 128
    }
  },
  "tokenizer": {
    "value": {
      "bos_token": "[BOS]",
      "eos_token": "[EOS]",
      "pad_token": "[PAD]",
      "unk_token": "[UNK]",
      "mask_token": "[MASK]",
      "output_dir": "interdiff/data/processed/zinc_tok_seqlen_128_vocabsize_500",
      "seq_length": 128,
      "vocab_size": 500,
      "use_selfies": false,
      "bos_token_id": 3,
      "dataset_path": "interdiff/data/zinc/zinc.txt",
      "eos_token_id": 4,
      "pad_token_id": 0,
      "unk_token_id": 1,
      "mask_token_id": 2
    }
  },
  "wandb_log": {
    "value": true
  },
  "experiment": {
    "value": {
      "wandb_project": "interdiff",
      "wandb_run_name": "ppo_qed_controllable_nlatents8_envs16_steps256_seed42"
    }
  },
  "from_scratch": {
    "value": false
  },
  "rl_train_cfg": {
    "value": {
      "budget": 10000000,
      "device": "cuda",
      "_target_": "interdiff.trainers.base_RL.RLTrainConfig",
      "ckpt_path": "ckpts/ppo_qed_controllable_nlatents8_envs16_steps256_seed42",
      "pad_token_id": 0,
      "log_frequency": 1,
      "tokenizer_dir": "interdiff/data/processed/zinc_tok_seqlen_128_vocabsize_500",
      "eval_frequency": 1,
      "n_mols_generate": 100,
      "always_save_checkpoint": false
    }
  }
}
</pre>
</summary>
</details>
